boa/contracts/vyper/vyper_contract.py

```diff
@@ -68,7 +68,98 @@
 # error detail where user possibly provided dev revert reason
 DEV_REASON_ALLOWED = ("user raise", "user assert")
 
+class ContractVerifier:
```

Comment:
imo this should be two separate functions (no need for a class) in a separate file. Maybe just add the functions to a dict so we can leverage a built-in KeyError for invalid explorer arguments like
```
contract_verifiers = {"etherscan": validate_etherscan, "blockscout": validate_blockscout}
```
but @charles-cooper might think different